testing/dart_isolate_runner.cc

```diff
@@ -47,110 +50,94 @@ AutoIsolateShutdown::~AutoIsolateShutdown() {
-      return;
+      return nullptr;
     }
 
     auto kernel_file = fml::OpenFile(kernel_file_path.c_str(), false,
                                      fml::FilePermission::kRead);
 
     if (!kernel_file.is_valid()) {
       FML_LOG(ERROR) << "Kernel file descriptor was invalid.";
-      return;
+      return nullptr;
     }
 
     auto kernel_mapping = std::make_unique<fml::FileMapping>(kernel_file);
 
     if (kernel_mapping->GetMapping() == nullptr) {
       FML_LOG(ERROR) << "Could not setup kernel mapping.";
-      return;
-    }
-
-    if (!root_isolate->get()->PrepareForRunningFromKernel(
-            std::move(kernel_mapping))) {
-      FML_LOG(ERROR)
-          << "Could not prepare to run the isolate from the kernel file.";
-      return;
+      return nullptr;
     }
-  } else {
-    if (!root_isolate->get()->PrepareForRunningFromPrecompiledCode()) {
-      FML_LOG(ERROR)
-          << "Could not prepare to run the isolate from precompiled code.";
-      return;
-    }
-  }
 
-  if (root_isolate->get()->GetPhase() != DartIsolate::Phase::Ready) {
-    FML_LOG(ERROR) << "Isolate is in unexpected phase.";
-    return;
+    settings.application_kernels = fml::MakeCopyable(
+        [kernel_mapping = std::move(kernel_mapping)]() mutable -> Mappings {
+          Mappings mappings;
+          mappings.emplace_back(std::move(kernel_mapping));
+          return mappings;
+        });
   }
 
-  if (!root_isolate->get()->Run(entrypoint, args,
-                                settings.root_isolate_create_callback)) {
-    FML_LOG(ERROR) << "Could not run the method \"" << entrypoint
-                   << "\" in the isolate.";
-    return;
+  auto isolate_configuration =
+      IsolateConfiguration::InferFromSettings(settings);
+
+  auto isolate =
+      DartIsolate::CreateRunningRootIsolate(
+          settings,                            // settings
+          vm_data->GetIsolateSnapshot(),       // isolate snapshot
+          std::move(task_runners),             // task runners
+          nullptr,                             // window
+          {},                                  // snapshot delegate
+          {},                                  // hint freed delegate
+          io_manager,                          // io manager
+          {},                                  // unref queue
+          {},                                  // image decoder
+          "main.dart",                         // advisory uri
+          entrypoint.c_str(),                  // advisory entrypoint
+          DartIsolate::Flags{},                // flags
+          settings.isolate_create_callback,    // isolate create callback
+          settings.isolate_shutdown_callback,  // isolate shutdown callback
+          entrypoint,                          // entrypoint
+          std::nullopt,                        // library
+          std::move(isolate_configuration)     // isolate configuration
+          )
+          .lock();
```

Comment:
You mean just return `std::shared_ptr`. I don't see why not. I like `weak_ptr` because the docs explicitly mention not to hold onto strong references to the isolate as that can interfere with the object from being collected if the isolate decides to kill itself. But if you have a strong opinion about this, I can update all callsites.

Reply:
If forgetting the `lock()` call could cause a bug, then I'd change it. If that is unlikely, then I don't have a strong opinion.

Reply:
Forgetting a lock would make the variable a weak pointer and that doesn't have a dereference operator. Calling any methods on the isolate would be impossible.